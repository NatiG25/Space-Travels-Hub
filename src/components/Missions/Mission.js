import React from 'react';
import PropTypes from 'prop-types';

const Mission = ({
  missionName,
  description,
  joinMissionHandler,
  cancelMissionHandler,
  reservation,
}) => {
  Mission.propTypes = {
    missionName: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    reservation: PropTypes.bool,
    joinMissionHandler: PropTypes.func.isRequired,
    cancelMissionHandler: PropTypes.func.isRequired,
  };

  Mission.defaultProps = {
    reservation: false,
  };

  return (
    <>
      <tr className="mission-row">
        <td>{missionName}</td>
        <td>{description}</td>
        <td className="status">
          {reservation
            ? <span className="status-field available-status">Active Member</span>

            : <span className="status-field unavailable-status">Not a Member</span> }
        </td>
        <td>
          {reservation
            ? (
              <button
                type="button"
                className="button-mission leave-mission"
                href="."
                onClick={cancelMissionHandler}
              >
                Cancel Mission
              </button>
            )
            : (
              <button
                type="button"
                className="button-mission join-mission"
                href="."
                onClick={joinMissionHandler}
              >
                Join Mission
              </button>
            )}
        </td>
      </tr>
    </>
  );
};

export default Mission;
