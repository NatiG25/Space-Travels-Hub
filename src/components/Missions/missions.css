.tableMissions,
th,
td {
  border: 1px solid black;
  border-collapse: collapse;
}

.reservation {
  color: black;
  padding: 10px;
  text-decoration: none;
  cursor: pointer;
  border-radius: 8px;
  border: 1px solid black;
}
