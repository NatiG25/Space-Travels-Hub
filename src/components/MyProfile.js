import React from 'react';

const MyProfile = () => (
  <>
    <h1>MyProfile</h1>
  </>
);

export default MyProfile;
