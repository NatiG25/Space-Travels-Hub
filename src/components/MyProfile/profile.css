.profileContainer {
  display: flex;
  flex-direction: row;
  width: 90%;
  height: auto;
  margin: 40px auto;
  gap: 50px;
  text-align: left;
}

h2 {
  margin-left: 40px;
  font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
}

.missionContainer,
.rocketContainer {
  width: 45%;
}

li {
  list-style-type: none;
  min-height: 30px;
  border: 2px solid #bcbcbc;
  margin: 20px auto;
  padding: 10px 0 0 10px;
}
