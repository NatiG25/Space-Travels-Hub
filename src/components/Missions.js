import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Mission from './Mission';
import { joinMission, cancelMission } from './redux/missions/missions';

const Missions = () => {
  const missionInfo = useSelector((state) => state.missionReducer);
  const dispatch = useDispatch();

  const joinMissionHandler = (id) => {
    dispatch(joinMission(id));
  };

  const cancelMissionHandler = (id) => {
    dispatch(cancelMission(id));
  };

  return (
    <main>
      <table className="tableMissions">
        <tbody>
          <tr>
            <th>Missions</th>
            <th>Description</th>
            <th>Status</th>
          </tr>
          {missionInfo.map((item) => (
            <Mission
              key={item.mission_id}
              missionName={item.mission_name}
              description={item.description}
              reservation={item.reserved}
              joinMissionHandler={() => joinMissionHandler(item.mission_id)}
              cancelMissionHandler={() => cancelMissionHandler(item.mission_id)}
            />
          ))}
        </tbody>
      </table>
    </main>
  );
};

export default Missions;
