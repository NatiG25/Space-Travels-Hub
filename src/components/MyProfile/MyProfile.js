import React from 'react';
import { useSelector } from 'react-redux';
import ProfileMission from './ProfileMission';
import './profile.css';

const MyProfile = () => {
  const reservedMissions = useSelector((state) => state.missionReducer);
  const rocketInfo = useSelector((state) => state.rocketReducer);
  const reserveRockets = rocketInfo.filter((rocket) => rocket.reserved);

  return (
    <div className="profileContainer">
      <div className="missionContainer">
        <h2>My Missions</h2>
        <ul>
          {reservedMissions
            .filter((data) => data.reserved)
            .map((filterData) => (
              <ProfileMission key={filterData.mission_id} missionName={filterData.mission_name} />
            ))}
        </ul>
      </div>
      <div className="rocketContainer">
        <h2>My Rockets</h2>
        <ul>
          {reserveRockets.map((rocket) => (
            <li key={rocket.id}>
              {rocket.rocket_name}
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default MyProfile;
