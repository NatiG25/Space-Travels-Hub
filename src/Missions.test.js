import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Missions from './components/Missions/Missions'
import MyProfile from './components/MyProfile/MyProfile'
import store from './components/redux/configureStore'

it('renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}><Missions /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
    const tree = renderer
      .create(<Provider store={store}><MyProfile /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
});
