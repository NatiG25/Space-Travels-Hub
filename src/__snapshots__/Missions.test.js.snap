// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main>
  <table
    className="tableMissions"
  >
    <tbody>
      <tr>
        <th>
          Missions
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
      </tr>
    </tbody>
  </table>
</main>
`;

exports[`renders correctly 2`] = `
Array [
  <h2>
    My Missions
  </h2>,
  <ul />,
]
`;
